ts/util/i18n/localizedString.ts

```diff
@@ -0,0 +1,303 @@
+import { getFallbackDictionary, getTranslationDictionary } from './translationDictionaries';
+import { en } from '../../localization/locales';
+import { getLocale } from './shared';
+import { LOCALE_DEFAULTS } from '../../localization/constants';
+import { deSanitizeHtmlTags, sanitizeArgs } from '../../components/basic/Localizer';
+import type { LocalizerDictionary } from '../../types/Localizer';
+
+type PluralKey = 'count';
+
+type ArgString = `${string}{${string}}${string}`;
+type RawString = ArgString | string;
+
+type PluralString = `{${PluralKey}, plural, one [${RawString}] other [${RawString}]}`;
+
+// type LocalizedString = string;
+
+type GenericLocalizedDictionary = Record<string, RawString | PluralString>;
+
+type TokenString<Dict extends GenericLocalizedDictionary> = keyof Dict extends string
+  ? keyof Dict
+  : never;
+
+// type GenericArgsRecord = Record<string, string | number>;
+
+/** The dynamic arguments in a localized string */
+type StringArgs<T extends string> =
+  /** If a string follows the plural format use its plural variable name and recursively check for
+   *  dynamic args inside all plural forms */
+  T extends `{${infer PluralVar}, plural, one [${infer PluralOne}] other [${infer PluralOther}]}`
+    ? PluralVar | StringArgs<PluralOne> | StringArgs<PluralOther>
+    : /** If a string segment follows the variable form parse its variable name and recursively
+       * check for more dynamic args */
+      T extends `${string}{${infer Var}}${infer Rest}`
+      ? Var | StringArgs<Rest>
+      : never;
+
+export type StringArgsRecord<T extends string> = Record<StringArgs<T>, string | number>;
+
+// TODO: move this to a test file
+//
+// const stringArgsTestStrings = {
+//   none: 'test',
+//   one: 'test{count}',
+//   two: 'test {count} second {another}',
+//   three: 'test {count} second {another} third {third}',
+//   four: 'test {count} second {another} third {third} fourth {fourth}',
+//   five: 'test {count} second {another} third {third} fourth {fourth} fifth {fifth}',
+//   twoConnected: 'test {count}{another}',
+//   threeConnected: '{count}{another}{third}',
```

Comment:
to remove ?

Reply:
@Aerilym let me know if still relevant or if I can remove those

Reply:
So this is actually a typescript text (sorta) AFAIK we dont really have a way to test types. ATM if you comment this out and theres a type error then the types have broken. My thoughts is either we leave it here and uncomment when we make changes to the types, or if we can, move it to a test, but as we're using mocha (instead of jest) the testing suite doesnt test types

Reply:
Resolving for now